### PR TITLE
Add multitest format

### PR DESCRIPTION
**Pull request by sskeirik**
Fixes: runtimeverification/firefly-michelson#6

### CODE REVIEW

**Comment by ehildenb** [comment]
Things to check for the super-long build time:

-   `z3 --version` >= 4.8.6. Check if Z3 is running for ridiculously long amounts of time.
-   Also takes forever to compile haskell backend?
-   Try `--heuristic oqbaR` : https://github.com/runtimeverification/algorand-sc-semantics/blob/master/Makefile#L245
-   If there's an obvious segmentation of rules, you can use `priority(...)` to tell the LLVM backend to manually partition the rules.

File: Makefile [hunk 1]
```diff
@@ -104,7 +104,8 @@ tangle_llvm    := k | concrete
 HOOK_NAMESPACES := TIME
 
 # NOTE: -W useless-rule check is quite expensive (increasing compilation times by several times), use -Wno useless-rule unless this check is needed!!!
-KOMPILE_OPTS += --hook-namespaces "$(HOOK_NAMESPACES)" --gen-bison-parser --emit-json -w all -Wno unused-symbol -Wno useless-rule
+export K_OPTS += -Xmx8G
+KOMPILE_OPTS += --hook-namespaces "$(HOOK_NAMESPACES)" --gen-glr-bison-parser -w all -Wno unused-symbol -Wno useless-rule
```
Comment (ehildenb) [inline Makefile:108]: GLR parser is on purpose?

File: common.md [hunk 2]
```diff
@@ -16,6 +16,98 @@ module MICHELSON-COMMON
...
+                                       accountNonce       : OperationNonce,
+                                       txnParam           : PreData,
+                                       txnAmount          : Mutez,
+                                       txnSource          : Address,
+                                       txnSender          : Address,
+                                       inputstack         : LiteralStack,
+                                       expected           : OutputStack,
+                                       pre                : BlockList,
+                                       post               : BlockList,
+                                       invs               : Map)
```
Comment (ehildenb) [inline common.md:42]: This is almost certainly the problem.

```
syntax TestContextStruct ::= "Context" "(" "..." "invs" ":" Map ")"
                            | "Context" "(" "..." "post" ":" BlockList ")"
                            | "Context" "(" "..." "post" ":" BlockList "invs" ":" Map ")"
                            ...
```

**Comment by ehildenb** [comment]
You can use configuration fragments instead for storing state, which you can see an example of in KEVM:

```
    syntax InternalOp ::= "#pushCallStack"
 // --------------------------------------
    rule <k> #pushCallStack => . ... </k>
         <callStack> (.List => ListItem(<callState> CALLSTATE </callState>)) ... </callStack>
         <callState> CALLSTATE </callState>

    syntax InternalOp ::= "#popCallStack"
 // -------------------------------------
    rule <k> #popCallStack => . ... </k>
         <callStack>  (ListItem(<callState> CALLSTATE </callState>) => .List) ... </callStack>
         <callState> _ => CALLSTATE </callState>

    syntax InternalOp ::= "#dropCallStack"
 // --------------------------------------
    rule <k> #dropCallStack => . ... </k>
         <callStack> (ListItem(_) => .List) ... </callStack>
```

**Review by nishantjr — COMMENTED** [review]
Looks good modulo the `TestName` change we discussed yesterday (i.e. use strings instead).

**Review by nishantjr — APPROVED** [review]
LGTM

**Pull request merged**